File: backend/src/main/java/com/awais/hr/module/succession/service/SuccessionService.java
package com.awais.hr.module.succession.service;

import java.util.List;
import java.util.Map;

public interface SuccessionService {
    List<Map<String, Object>> getPositions();
    List<Map<String, Object>> getSuccessionPlans();
    void addPosition(Map<String, Object> body);
    void addSuccessorToPlan(Map<String, Object> body);
    List<Map<String, Object>> getTalentPools();
    void addTalentPool(Map<String, Object> body);
    void addMemberToPool(Map<String, Object> body);
}

File: frontend/src/services/successionService.js
import apiClient from './api';

export const getPositions = () => apiClient.get('/succession/positions');
export const addPosition = (dto) => apiClient.post('/succession/positions', dto);
export const getSuccessionPlans = () => apiClient.get('/succession/plans');
export const addSuccessorToPlan = (dto) => apiClient.post('/succession/plans', dto);
export const getTalentPools = () => apiClient.get('/succession/talent-pools');
export const addTalentPool = (dto) => apiClient.post('/succession/talent-pools', dto);
export const addMemberToPool = (dto) => apiClient.post('/succession/talent-pools/members', dto);
